metrics/compute/compute.go

```diff
@@ -37,8 +38,9 @@ func OkOrPassesAndUnknownOrPasses(status *metrics.CompleteTestStatus) bool {
 
 // Gather results from test runs into input format for Compute* functions in
 // this module.
-func GatherResultsById(allResults *[]metrics.TestRunResults) (
+func GatherResultsById(ctx context.Context, allResults *[]metrics.TestRunResults) (
 	resultsById TestRunsStatus) {
+	logger := ctx.Value(shared.DefaultLoggerCtxKey()).(shared.Logger)
```

Comment:
Fall back to a nil logger? Maybe `shared.GetLoggerOrNil(ctx)` or similar would help.